### ISSUE TITLE
Windows 7 taskbar pinned shortcut compatibility

### ISSUE THREAD

**Issue by GoogleCodeExporter** [mintty/mintty#420]
```
I have two pinned mintty:
"C:\cygwin\bin\mintty.exe -- ssh linux" with icon gentoo.ico, let's say it's 
gentoo-mintty
"C:\cygwin\bin\mintty.exe -" with default icon, let's say it's cygwin-mintty

before starting gentoo-mintty, the two icon of pinned shortcut looks just fine. 
but after starting the gentoo-mintty, the icon becomes default icon instead of 
the custom one (gentoo.ico)
-> suggested enhancement 1: adds support for custom app icon from command line

close all mintty, start the cygwin-mintty, taskbar is shown as
[gentoo-mintty (not started)] | [cygwin-mintty (started)]
but after i press alt+F2 (the "New" window feature of mintty), taskbar looks 
like
[gentoo-mintty (started)] | [cygwin-mintty (started)], this is unexpected
it's expected to be
[gentoo-mintty (not started)] | [cygwin-mintty (started)][cygwin-mintty 
(started)] (the two should group next to each other)

it looks like the "New" window mintty is started with command line
"C:\cygwin\bin\mintty.exe" instead of "C:\cygwin\bin\mintty.exe -", which make 
window cannot recognize what this task should group, so it find the first one 
that match only the path part of command line without arguments

-> suggested enhancement 2: when using alt+F2/New feature, fork new mintty with 
exactly same command line argument as current one (don't forget to load correct 
custom icon btw)
```

Original issue reported on code.google.com by `xue...@gmail.com` on 15 Feb 2014 at 4:23

**Comment by dscho** [comment]
> suggested enhancement 1: adds support for custom app icon from command line

Is that not addressed by [the `-icon` command line option](https://github.com/mintty/mintty/blob/271b95218844eac5ba7b85fee3b0124864d099e5/docs/mintty.1#L77-L81)?

**Comment by mintty** [comment]
Issue 1: As dscho noted, the ```-i``` or ```--icon``` option would achieve the desired effect; however, it would be more intuitive and desirable if a dedicated icon configured for the shortcut being pinned would also persist on the taskbar. This is not the case, probably because mintty sets its own icon with the Windows API. A patch that handles this nicely (while keeping the default behaviour) would be appreciated.

**Comment by mintty** [comment]
Issue 2: I cannot reproduce this one. If I pin two different invocations of mintty with two different icons and start a few instances, behaviour is as expected here.

**Comment by mintty** [comment]
Issue 1: On retesting, an icon supplied with -i does not get persisted in a pinned shortcut.

**Comment by mintty** [comment]
Issue 2: It should help to use options like -o AppID=mintty.gentoo to distinguish the various shortcuts and achieve appropriate grouping.

**Comment by mintty** [comment]
Mintty could implicitly derive different AppIDs corresponding to the icon file base name for those invocations that explicitly select an icon, based on the assumption that people setting different icons may wish to group instances accordingly...
Comments welcome.

**Comment by mintty** [comment]
With release 2.2.1, the issues are supposedly resolved as far as possible.
Further comments will be considered, though.

**Comment by mintty** [comment]
Having understood more of the insane jungle of Windows shortcuts, app ids, and taskbar pinning, issue 2 (Alt+F2) is recognised. There is a solution using the AppId mechanism but it involves using an external tool to attach an AppId to a shortcut (to be pinned and then behave consistently).
Actually a hint to this used to be included in the mintty manual page. I will reinsert it and add instructions to the wiki hints page.

**Comment by mintty** [comment]
Release 2.2.3: Deriving icon from desktop shortcut.
I hope this issue has been addressed meanwhile sufficiently. Otherwise respond here for reopening.

**Comment by mintty** [comment]
Released 2.9.4 with improved taskbar icon grouping behaviour and configuration.
